test/strings/basic.jl

```diff
@@ -938,6 +938,17 @@ end
     end
 end
 
+@testset "Conversion to Type{Union{String, SubString{String}}}" begin
+    str::String = "abc"
+    substr = SubString(str, 1, 3)
+    for T in [typeof(str), typeof(substr)]
```

Comment:
Might as well write this out explicitly:

```suggestion
    for T in [String, SubString{String}]
```

Reply:
Okay.